src/app/views/river-detail/flow-tab/components/gage-chart-controls.vue

```diff
@@ -224,13 +227,17 @@ export default {
       this.$store.dispatch(metricsActions.FETCH_GAGE_METRICS, this.$route.params.id)
     },
     async fetchReadings () {
-
       this.$emit('gage-change', this.formData.gauge_id)
       await this.setTimeScale()
+      const gageInfoPromise = this.fetchGaugeInfo()
       this.$store.dispatch(
         readingsActions.FETCH_GAGE_READINGS_DATA,
         this.formData
       )
+      await gageInfoPromise
+    },
+    async fetchGaugeInfo () {
+      this.$store.dispatch(gaugeInfoActions.FETCH_GAGE_INFO, this.formData.gauge_id)
     }
```

Comment:
This seems odd. Why not just call the dispatch directly on L37? `await this.$store.dispatch`

Reply:
yeah I though that was going to be a call that was going to be used more than once, and I don't really like having `await this.$store.dispatch` sprinkled all over.

Reply:
@mattanger True that. `mapActions` might help. 

```js
methods: {
  ...mapActions({
    loadGauge: 'GaugeInfoSubModule/FETCH_GAGE_INFO'
  })
}
```

Cant remember what the submodule is called. or how crazy nested it is... 

Then you could just call:

```js
this.loadGauge(this.formData.gauge_id)
```

A little nicer.